hstream-io/HStream/IO/Worker.hs

```diff
@@ -177,6 +178,33 @@ recoverTask worker@Worker{..} name = do
       let newConnCfg = J.insert "hstream" (J.toJSON hsConfig) connectorConfig
       createIOTaskFromTaskInfo worker taskId taskInfo{connectorConfig=newConnCfg} options True False False
 
+-- update config and restart
+alterConnectorConfig :: Worker -> T.Text -> T.Text -> IO ()
+alterConnectorConfig worker name config = do
+  updated <- updateConnectorConfig worker name config
+  when updated $ do
+    Log.info $ "updated connector config, connector:" <> Log.build name
+    E.catch
+      (stopIOTask worker name True)
+      (\(e :: E.SomeException) -> Log.warning $ "failed to stop io task:" <> Log.buildString (show e))
+    Log.info $ "paused connector:" <> Log.build name
+    recoverTask worker name
+    Log.info $ "resumed connector:" <> Log.build name
+
+updateConnectorConfig :: Worker -> T.Text -> T.Text -> IO Bool
+updateConnectorConfig worker name config = do
+  Log.info $ "updating connector config, connector:" <> Log.build name <> ", overrided:" <> Log.build config
```

Comment:
I think we can remove this log because we have log all messages we need in the handler function